app/src/main/java/com/internship/move/feature/authentication/LoginFragment.kt

```diff
@@ -16,11 +17,21 @@ class LoginFragment : Fragment(R.layout.fragment_login) {
         super.onViewCreated(view, savedInstanceState)
 
         binding.homeBtn.setOnClickListener {
+            updateSharedPreferences()
             findNavController().navigate(LoginFragmentDirections.actionLoginFragmentToMapGraph())
         }
 
         binding.registerBtn.setOnClickListener {
             findNavController().navigate(LoginFragmentDirections.actionLoginFragmentToRegisterFragment())
         }
     }
+
+    private fun updateSharedPreferences() {
+        val sharedPref = requireActivity().getSharedPreferences(
+            getString(R.string.preference_file_key), Context.MODE_PRIVATE) ?: return
```

Comment:
I know that in the official docs it shows that the keys are stored in strings.xml. That's a good solution, and I am sure many projects are implemented like this. Although, in our team we prefer having these keys as constants, either in a companion object or in a Constants object SharedPrefKeys in another file (or another solution we'll talk about later)

Since they aren't user facing strings, they won't require translation and we'll have to write extra values in other languages string files and make sure they are ignored at translation (get's a bit more complex to do if you also use a remote translation tool)
That being said, there are some keys that we store in strings.xml files, this is a case by case rule.

in Constants.kt we'd have something like:
```
object Constants {

    object SharedPref {
        const val KEY_APP_PREFERENCES = "com.internship.move.KEY_APP_PREFERENCES"

        const val KEY_HAS_VISITED_AUTHENTICATION = "KEY_HAS_VISITED_AUTHENTICATION"
        // other keys. We have the same name and value for a key just for convenience, so we can copy paste it 
        //quickly
    }
}
```

Reply:
Also regarding the [shared preferences file key value](https://developer.android.com/training/data-storage/shared-preferences#GetSharedPreferences)
"When naming your shared preference files, you should use a name that's uniquely identifiable to your app. An easy way to do this is prefix the file name with your [application ID](https://developer.android.com/studio/build/configure-app-module#set_the_application_id). For example: "com.example.myapp.PREFERENCE_FILE_KEY" "